### DISCUSSION TITLE
How to exclude a directory from being checked?

### DISCUSSION THREAD

**Discussion by Imipenem** [agronholm/typeguard#213] [Q&A]
Hey,

I'm currently using typeguard for one of my projects. I'm using it via the `pytest plugin` and using `nox` to actually "run" it.

My code is like this:
```
@session(python=python_versions)
def typeguard(session: Session) -> None:
    """Runtime type checking using Typeguard."""
    session.install(".")
    session.install("pytest", "typeguard", "pygments", "ruamel.yaml")
    session.run("pytest", f"--typeguard-packages={package}", *session.posargs)
```
    
  (So basically only the last line is the command that really gets called with typeguard checking). 

I want to know, if it's possible to **exclude an entire directory from being checked**, for example via a command line parameter and not by editing any code in the corresponding directory?

Many thanks in advance ;)

**Comment by agronholm** [comment]
Sorry I missed this issue. Not sure why you added an "enhancement" label when this is clearly a question.

Typeguard does not operate based on directories, but modules (and I assume you are talking about the import hook).
Currently, the pytest plugin only allows specifying the list of packages to be instrumented, so there is no fine grained control. This is something I'm looking to add in v3.0 if possible.

**Reply by agronholm** [reply]
Do you need further information or does this answer your question?

**Reply by Imipenem** [reply]
Sry for the late response. Yes I dont have any questions anymore. 

Thank you
